src/app/models/form.server.model.ts

```diff
@@ -631,6 +632,26 @@ const compileFormModel = (db: Mongoose): IFormModel => {
     ).exec()
   }
 
+  // Updates specified form logic.
+  FormSchema.statics.updateFormLogic = async function (
+    this: IFormModel,
+    formId: string,
+    logicId: string,
+    updatedLogic: ILogicSchema,
+  ): Promise<IFormSchema | null> {
+    return this.findByIdAndUpdate(
+      mongoose.Types.ObjectId(formId),
```

Comment:
no need for this, you can leave `formId` as a string

Reply:
edited